### DISCUSSION TITLE
How can I make Avalonia act as a VNC server?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13577] [Q&A] [answered]
Recently I found this video:
https://www.youtube.com/watch?v=HAyHIvSzweM&ab_channel=NikitaTsukanov

I have been coding an embedded Linux program (by Avalonia 11.0.5) for embedded devices these days and I am thinking about how the user can control it with PC (windows platform). 

This is a bit different from traditional cross-platform, because I don't want to expose the functions of the embedded hardware as an interface to connect with programs on Windows.

It seems we can control an Avalonia program directly via VNC Viewer in the video above.

I looked up how to do it on Google, but I only saw a lot of people saying they did it, but no tutorials.

And I found a related Nuget library (https://www.nuget.org/packages/Avalonia.Headless.Vnc/), but I don't understand how to use it.

How can I achieve this? Are there any tutorials? Thank you.

**Comment by maxkatz6** [comment, marked as answer]
Yes, we have VNC server, yes https://github.com/AvaloniaUI/Avalonia/blob/master/src/Headless/Avalonia.Headless.Vnc/HeadlessVncPlatformExtensions.cs#L37-L40

VNC backend initially was a simple proof of concept, it wasn't much documented, but you can use it for sure.
From my memory, the only sample is StartWithHeadlessVncPlatform usage in the ControlCatalog: https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.NetCore/Program.cs#L59C32-L59C60

**Reply by kyuranger** [reply]
Thanks a lot, let me try this. 

In addition, it would be better if Avalonia team could add relevant introduction to Avalonia's tutorial, not just to teach others how to use it, many people don't even know that Avalonia has this function.

**Comment by pedroveiga99** [comment]
@kyuranger Did you manage to do it? I think I'm in a similar situation to you. I did not manage to make it work. Also, did you want to show the app in both the embedded hardware and VNC? I want to do that but according to this this discussion https://github.com/AvaloniaUI/Avalonia/issues/4608 it might not be possible...

**Reply by kyuranger** [reply]
Yes, I want to do the same as you, but then I never tried maxkatz6's way, and my LINUX DRM program is still not finished.
